lib_i2s/src/i2s_frame_slave_impl.h

```diff
@@ -43,16 +47,23 @@ static void i2s_frame_slave0(client i2s_frame_callback_if i2s_i,
     // Since #pragma unsafe arrays is used need to ensure array won't overflow.
     assert((num_in << 1) <= 16);
 
+    const unsigned data_bit_offset = 32 - num_data_bits;
+    const unsigned data_bit_mask = UINT_MAX >> data_bit_offset; // e.g. 00011111 for 5b data
+    
+    i2s_frame_slave_init_ports(p_dout, num_out, p_din, num_in, p_bclk, p_lrclk, bclk);
 
-    while(1){
-        i2s_frame_slave_init_ports(p_dout, num_out, p_din, num_in, p_bclk, p_lrclk, bclk);
 
+    while(1)
+    {
```

Comment:
Regarding reformatting code (eg. newline before open brace), the general rule is to keep the existing style, even if not personal preference. There are some rough coding style guidelines floating around somewhere but they aren't widely used..